src/OpenCL.cpp

```diff
@@ -172,14 +180,15 @@ __kernel void out_transform(__global float *M, __global net_t *Y,
                     temp_m[2*4 + 1] + temp_m[2*4 + 2] + temp_m[2*4 + 3] -
                     temp_m[3*4 + 1] + temp_m[3*4 + 2] + temp_m[3*4 + 3];
 
```

Comment:
Would calculating these only if they are needed, within the following ifs save anything?

Reply:
We can do better here, but to late for I go to bed. look at it tomorrow.

Reply:
I already tested that a while ago and I saw no gain.

On GPUs you're often forced to run all code in all ifs, even if some get discarded/masked, so these kind of tricks almost never works.